keras_hub/src/utils/tensor_utils.py

```diff
@@ -21,17 +21,25 @@
 NO_CONVERT_COUNTER = threading.local()
 
 
-def pad(x, shape, padding_side, pad_value):
-    if padding_side == "left":
+def pad(x, shape, padding_side, pad_value, axis=-1):
+    if padding_side == "left" and pad_value is not None:
```

Comment:
I don't fully understand what we are trying to do here, but I think it is buggy, we should go back to the reverse and `to_tensor` approach and avoid this manual padding.